File: models/sign-record.js
var errorCodes = require('../services/error-codes').errorCodes;

var SignRecord = {
  signId: {
    type: String,
    required: true
  },
  phoneId: {
    type: String,
    required: true
  },
  studentId: {
    type: String,
    required: true
  },
  studentName: {
    type: String
  },
  studentAvatar: {
    type: String
  },
  distance: {
    type: Number
  },
  state: {
    type: Number,
    enum: {
      values: [0, 1, 2],
      message: errorCodes.SignRecordStateError
    }
  },
  type: {
    type: Number,
    enum: {
      values: [0, 1],
      message: errorCodes.SignRecordTypeError
    }
  },
  createdAt: {
    type: String
  },
  battery: {
    type: Number
  }
};

 module.exports = SignRecord;

File: app/src/main/java/com/scnu/zhou/signer/component/bean/sign/SignRecord.java
package com.scnu.zhou.signer.component.bean.sign;

/**
 * Created by zhou on 16/10/22.
 */
public class SignRecord {

    private String id;
    private String signId;
    private String phoneId;
    private String studentId;
    private String studentName;
    private String studentAvatar;
    private int distance;
    private int state;
    private int type;
    private String createdAt;
    private int battery;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getSignId() {
        return signId;
    }

    public void setSignId(String signId) {
        this.signId = signId;
    }

    public String getPhoneId() {
        return phoneId;
    }

    public void setPhoneId(String phoneId) {
        this.phoneId = phoneId;
    }

    public String getStudentId() {
        return studentId;
    }

    public void setStudentId(String studentId) {
        this.studentId = studentId;
    }

    public String getStudentName() {
        return studentName;
    }

    public void setStudentName(String studentName) {
        this.studentName = studentName;
    }

    public String getStudentAvatar() {
        return studentAvatar;
    }

    public void setStudentAvatar(String studentAvatar) {
        this.studentAvatar = studentAvatar;
    }

    public int getDistance() {
        return distance;
    }

    public void setDistance(int distance) {
        this.distance = distance;
    }

    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public int getBattery() {
        return battery;
    }

    public void setBattery(int battery) {
        this.battery = battery;
    }
}
